how-to/security/openssh-server.md

```diff
@@ -65,35 +67,61 @@ Many other configuration directives for `sshd` are available to change the serve
 
 ## SSH keys
 
-SSH allows authentication between two hosts without the need of a password. SSH key authentication uses a **private key** and a **public key**.
+SSH allows authentication between two hosts without the need of a password, using cryptographic keys instead.
+{term}`SSH-key` authentication uses a **private key** and a **public key**.
+
+We recommend the **ed25519 Elliptic Curve algorithm** (`-t ed25519`) due to shorter keysize and lower computational requirements.
+Alternatively, you can create a key using the **RSA Algorithm** (`-t rsa -b 4096`, for 4096 bit keysize) instead.
 
-To generate the keys, run the following command:
+To generate a key pair, run the following command:
 
 ```bash
-ssh-keygen -t rsa
+ssh-keygen -t ed25519
 ```
 
-This will generate the keys using the **RSA Algorithm**. At the time of this writing, the generated keys will have 3072 bits. You can modify the number of bits by using the `-b` option. For example, to generate keys with 4096 bits, you can use:
+During creation, you will be prompted for a key password, which you would enter to use that key (every time, or instead using an [ssh agent](#ssh-agent).
+
+By default, the public key is saved in the file `~/.ssh/id_<algorithm>.pub`, while `~/.ssh/id_<algorithm>` is the private key.
+To allow login for a user via key, append the content of `id_ed25519.pub` (`id_rsa.pub` for RSA) to `target_machine:~/.ssh/authorized_keys` by running:
 
 ```bash
-ssh-keygen -t rsa -b 4096
+ssh-copy-id username@remotehost
 ```
 
-During the process you will be prompted for a password. Simply hit <kbd>Enter</kbd> when prompted to create the key.
-
-By default, the public key is saved in the file `~/.ssh/id_rsa.pub`, while `~/.ssh/id_rsa` is the private key. Now copy the `id_rsa.pub` file to the remote host and append it to `~/.ssh/authorized_keys` by running:
+Finally, double check the permissions on the `authorized_keys` file -- only the authenticated user must have write permissions.
+If the permissions are not correct then change them by:
 
 ```bash
-ssh-copy-id username@remotehost
+chmod go-w .ssh/authorized_keys
 ```
 
-Finally, double check the permissions on the `authorized_keys` file -- only the authenticated user should have read and write permissions. If the permissions are not correct then change them by:
+You should now be able to SSH to the `target_machine` without being prompted for a password.
+
+To troubleshoot this, have a look at the `target_machine`'s journal:
 
 ```bash
-chmod 600 .ssh/authorized_keys
+sudo journalctl -eu sshd.service
+```
+
+Since you can have multiple {term}`SSH-key`s, you can configure which to use for which target machine in `.ssh/config` (see [`man ssh_config`](https://manpages.ubuntu.com/manpages/latest/en/man5/ssh_config.5.html)).
```

Comment:
Missing `~`: `~/.ssh/config`

And for the manpage, can we use the new manpage reference scheme here? I forgot the syntax, but should be similar to the `{term}` one.